docs/federated_managed_identity.md

```diff
@@ -0,0 +1,133 @@
+# Federated Managed Identity (FMI)
+
+## Overview
+
+Federated Managed Identity (FMI) enables Azure services to use federated credentials for authentication via an assertion callback. The FMI credential is typically obtained from a Resource Management API (RMA) service and used with the special client ID `urn:microsoft:identity:fmi`.
+
+## Key Features
+
+- **Assertion-Based Authentication**: Uses callback functions to dynamically retrieve FMI credentials
+- **Cache Isolation**: Tokens acquired with different FMI paths are cached separately
+- **FMI Path Support**: The `WithFMIPath` option identifies the specific federated credential path
+
+## Basic Usage
+
+### Step 1: Create an FMI Credential Provider
+
+```go
+import (
+    "context"
+    "github.com/AzureAD/microsoft-authentication-library-for-go/apps/confidential"
+)
+
+// GetFMICredentialFromRMA acquires an FMI token from the RMA service
+func GetFMICredentialFromRMA(ctx context.Context) (string, error) {
+    // Get your certificate for authenticating to RMA
+    cert, privateKey, err := getCertificateData()
+    if err != nil {
+        return "", err
+    }
+
+    // Create credential from certificate
+    cred, err := confidential.NewCredFromCert(cert, privateKey)
+    if err != nil {
+        return "", err
+    }
+
+    // Create client for RMA service
+    rmaClient, err := confidential.New(
+        "https://login.microsoftonline.com/tenant-id",
+        "your-rma-client-id",
+        cred,
+    )
+    if err != nil {
+        return "", err
+    }
+
+    // Acquire FMI token from RMA with FMI path
+    result, err := rmaClient.AcquireTokenByCredential(
+        ctx,
+        []string{"api://AzureFMITokenExchange/.default"},
+        confidential.WithFMIPath("YourFmiPath/CredentialPath"),
+    )
+    if err != nil {
+        return "", err
+    }
+
+    return result.AccessToken, nil
+}
+```
+
+### Step 2: Use FMI with Assertion Callback
```

Comment:
"step 2" implies this code should run after "step 1" but in fact they're alternatives